src/main/io/osd.c

```diff
@@ -997,4 +997,12 @@ void osdUpdate(timeUs_t currentTimeUs)
     }
 #endif
 }
+
+void osdForceRedraw(void)
```

Comment:
I think this function whould be called `osdInitiateFullRedraw` and should only set a flag. Actual activity should happen from OSD code itself.

Reply:
That would require setting a flag somewhere and checking it every iteration. Not sure if that would affect draw performance, but we'd need to check that flag every time.

Note that this is only called from MSP when the OSD config is altered, so we shouldn't expect that to happen during a flight. If we ever really need an async version, we could add it later, but it doesn't seem necessary right now.

However, async is the more correct option so if you can clarify whether you're sure we can spare the cycles I'll make the changes.

Reply:
Ok, let's keep it simple way for now.

Reply:
@digitalentity After reading the code for display port handling and the MAX746 driver, I think it's better to redo it as you suggested. This way could cause concurrency related issues. It does not at the moment due to the way the MAX7456 implementation is done, but the external display port API provides no guarantees to do so.

Reply:
@digitalentity Just sent the reworked version. Please, merge it if everything's fine.